File: CS-Examples/10_Shapes/GroupShapes.vb
Imports Spire.Xls
Imports Spire.Xls.Core.MergeSpreadsheet.Collections
Imports Spire.Xls.Core
Imports System.ComponentModel
Imports System.Text

Namespace GroupShapes
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Get the first worksheet from the workbook
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Add a rounded rectangle shape to the worksheet and set its properties
            Dim shape1 As IPrstGeomShape = worksheet.PrstGeomShapes.AddPrstGeomShape(1, 3, 50, 50, PrstGeomShapeType.RoundRect)

            ' Add a triangle shape to the worksheet and set its properties
            Dim shape2 As IPrstGeomShape = worksheet.PrstGeomShapes.AddPrstGeomShape(5, 3, 50, 50, PrstGeomShapeType.Triangle)

            ' Get the group shape collection from the worksheet
            Dim groupShapeCollection As GroupShapeCollection = worksheet.GroupShapeCollection

            ' Group the specified shapes together
            groupShapeCollection.Group(New Spire.Xls.Core.IShape() {shape1, shape2})

            ' Specify the output file name for saving the modified workbook
            Dim result As String = "GroupShape.xlsx"

            ' Save the workbook to the specified output file path in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/10_Shapes/GroupShapes.cs
using Spire.Xls;
using Spire.Xls.Core.MergeSpreadsheet.Collections;
using Spire.Xls.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace GroupShapes
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Get the first sheet from the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            // Add shapes to the worksheet
            IPrstGeomShape shape1 = worksheet.PrstGeomShapes.AddPrstGeomShape(1, 3, 50, 50, PrstGeomShapeType.RoundRect);
            IPrstGeomShape shape2 = worksheet.PrstGeomShapes.AddPrstGeomShape(5, 3, 50, 50, PrstGeomShapeType.Triangle);

            // Group the shapes
            GroupShapeCollection groupShapeCollection = worksheet.GroupShapeCollection;
            groupShapeCollection.Group(new Spire.Xls.Core.IShape[] { shape1, shape2 });

            // Save the workbook to a file
            string result = "GroupShape.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}
